### PR TITLE
(baby-project) adds CI support

### PR DESCRIPTION
**Pull request by dusan-turajlic**

### CODE REVIEW

File: .github/workflows/gradle.yml
Comment (AndreasNasman) [inline .github/workflows/gradle.yml]:
```suggestion
    branches: [ main ]
```

File: .github/workflows/gradle.yml
Comment (AndreasNasman) [inline .github/workflows/gradle.yml]:
```suggestion
    branches: [ main ]
```

File: .github/workflows/gradle.yml
Comment (AndreasNasman) [inline .github/workflows/gradle.yml]: Could we use a newer version of Java or are there some limitations?

**Comment by AndreasNasman** [comment]
My main concern is the Java version. I think we should try to use the newest stable version if possible. According to [Wikipedia](https://en.wikipedia.org/wiki/Java_version_history), Java 17 will be LTS in September, and going from 11 to 17 feels like quite a big jump if we later decide to upgrade.

**Comment by dusan-turajlic** [comment]
I agree! But I still think using the latest LTS is safer.

I can’t say if there are any pros or cons in using java 15 but let's evaluate if we should. A benefit would then be that we would have an easier time upgrading to 17.

But i guess the con is we need to deal with java 15 specific problems.

I’m not for or against either but let's check the current state and what others have experienced before merging.

Ether way great point to bring up 👍

**Comment by dusan-turajlic** [comment]
> If you are making sure to use the very latest IDEs, frameworks and build tools and starting a greenfield project, you can, without hesitation, use Java 11 (LTS) or even the latest Java 15.

From [here](https://www.marcobehler.com/guides/a-guide-to-java-versions-and-features)

Based on the info there we don't realy gain anything by using Java 15 because we are using kotlin to write the code.

The Java release cycle has become alot more rapid, its now every 6 mounths so the jump from 11 to 17 is not as big as one might think.

But ether way It seems like we can totaly use 15. Maybe we should!

This is a Greengield project and why not use the latest stuff.

It's a risk either way, ether things don't work on 15 and we have to downgrade to 11 or we use 11 and upgrading to 17 is a pain!

IMO, let's just use 15.

**Comment by AndreasNasman** [comment]
Yeah, version 15 sounds good! 👍

Since the backend will be written Kotlin, is the Java version specified here only for the CI process?

**Comment by dusan-turajlic** [comment]
@AndreasNasman I updated the Java version and added the a `.java-version` to the project so if you use jenv it will automaticly pick up and use Java 15

**Comment by dusan-turajlic** [comment]
Note there is a downside with not running LTS (I relised now). Things like [JRebel](https://www.jrebel.com/products/jrebel) or [HotSwap](http://hotswapagent.org/) won't work with Java 15. But it would be good for longterm developement confert to use something like that. But lets merge this after the review is done so we can get started on the developemnt!

**Review by AndreasNasman — APPROVED** [review]
Looking good! 👍

**Pull request merged**
